npm/cmd/root.go

```diff
@@ -12,6 +19,27 @@ func NewRootCmd() *cobra.Command {
 		CompletionOptions: cobra.CompletionOptions{
 			DisableDefaultCmd: true,
 		},
+		PersistentPreRunE: func(cmd *cobra.Command, args []string) error {
```

Comment:
why move this from the start cmd? Also has some impacts like to npm/Dockerfile, so might be better to scope to another PR?

Reply:
What Dockerfile changes do you anticipate @huntergregory ?

Reply:
perhaps?
https://github.com/Azure/azure-container-networking/blob/d6ba4b9e363cbd394cb544753f7bb0b31fdcf9dc/npm/Dockerfile#L33

Reply:
So are you suggesting we update this to "controller"/"daemon"? The default behavior, as for now, should not use fan-out IMO.

Reply:
Took another look at the code, and this makes sense now.